### DISCUSSION TITLE
Is it good practice to use a AppService with another AppService

### DISCUSSION THREAD

**Discussion by vnetonline** [abpframework/abp#8565] [Q&A]
I was wondering is it good practice to use a AppService within another AppService or should you use IReposotries and Managers like the RoleManager etc etc

**Comment by amsaighi** [comment]
If you use appservice with another you will create a dependency between them. For a good practice try to handle hem with repository in appService or with Manager in your domain.

AppService : 

https://docs.abp.io/en/abp/latest/Domain-Driven-Design-Implementation-Guide#application-services

Example : 

https://docs.abp.io/en/abp/latest/Domain-Driven-Design-Implementation-Guide#examples

**Reply by muratyuceer** [reply]
maybe you can use domain service

**Comment by ebicoglu** [comment]
use a DomainService that gathers both or as @amSaighi says create a Manager class
